Made (Java)ExecFork implement ProcessForkOptions

DreierF opened the pull request:
Hey there,

the pull request makes the two tasks implement Gradle's ProcessForkOptions interface. This allows us to attach JaCoCo to a JavaExecFork task. Furthermore this fixes #8.
I also added the annotation for #14.

To be consistent with the `ProcessForkOptions` interface I removed the commandLine property in favour of `executable`.

Would be really nice if we could get this merged :)

psxpaul commented on sample_projects/shell_script/build.gradle:
```diff
@@ -18,7 +18,7 @@ task verify(dependsOn: 'startDaemon') {
 }
 
 task startDaemon(type: com.github.psxpaul.task.ExecFork) {
-  commandLine = './com/github/psxpaul/example/Main.sh'
```
It would be nice to add a setter for this property, marked as deprecated, to preserve backwards compatibility.

DreierF replied:
Done

DreierF commented:
Sorry it took me so long to address the review. I re-added the explicit types as suggested.
Adding a sample project was indeed a good idea, because it revealed some more problems:
- up-to-date check was no longer properly working when applying jacoco (see comment in code)
- no coverage was generated, because the forked process was killed instead of terminated, which did not trigger JaCoCo's shutdown hooks to write coverage to disk

Both are fixed now 🙂 

I also updated gradle to 5.0 and adjusted the plugin to use the new API's where necessary. So the plugin will now require Gradle 4.10.0 or newer.

DreierF commented:
Any chance you could have a look at the changes?

DreierF commented:
Thanks a lot 🙂
